libs/ast_generic/AST_generic.ml

```diff
@@ -548,12 +548,15 @@ and qualifier =
 (*****************************************************************************)
 and id_info = {
   id_resolved : resolved_name option ref;
+      [@equal
```

Comment:
why do we need to do anything special here? Why the default autogenerated eq does not work here?